openmc/waste.py

```diff
@@ -0,0 +1,279 @@
+        _waste_disposal_rating(mat, 'NRC_short_C', metal=metal),
+    ]
+
+    # Determine which nuclides are present in Table 1 and Table 2
+    table1_nuclides_present = (ratio1 > 0.0)
+    table2_nuclides_present = any(x > 0.0 for x in ratio2)
+
+    # Helper function for classifying based on Table 2
+    def classify_table2(col1, col2, col3):
+        if col1 < 1.0:
+            return "Class A"
+        elif col2 < 1.0:
+            return "Class B"
+        elif col3 < 1.0:
+            return "Class C"
+        else:
+            return "GTCC"
+
+    if table1_nuclides_present and table2_nuclides_present:
+        # Classification based on §61.55(a)(5)
+        if ratio1 < 0.1:
+            return classify_table2(*ratio2)
+        elif ratio1 < 1.0:
+            return "Class C" if ratio2[2] < 1.0 else "GTCC"
+        else:
+            return "GTCC"
+
+    elif table1_nuclides_present:
+        # Classification based on §61.55(a)(3)
+        if ratio1 < 0.1:
+            return "Class A"
+        elif ratio1 < 1.0:
+            return "Class C"
+        else:
+            return "GTCC"
+
+    elif table2_nuclides_present:
+        # Classification based on §61.55(a)(4)
+        return classify_table2(*ratio2)
+
+    else:
+        # Classification based on §61.55(a)(6)
+        return "Class A"
+
+
+def _waste_disposal_rating(
+        mat: openmc.Material,
+        limits: str | dict[str, float] = 'Fetter',
+        metal: bool = False,
+    ) -> float:
+    """Return the waste disposal rating for a material.
+
+    This method returns a waste disposal rating for the material based on a set
+    of specific activity limits. The waste disposal rating is a single number
+    that represents the sum of the ratios of the specific activity for each
+    radionuclide in the material against a nuclide-specific limit. A value less
+    than 1.0 indicates that the material "meets" the limits whereas a value
+    greater than 1.0 exceeds the limits.
+
+    Parameters
+    ----------
+    mat : openmc.Material
+        The material to classify.
+    limits : str or dict, optional
+        The name of a predefined set of specific activity limits or a dictionary
+        that contains specific activity limits for radionuclides, where keys are
+        nuclide names and values are activities in units of [Ci/m3]. The
+        predefined options are:
+
+        - 'Fetter': Uses limits from Fetter et al. (1990)
+        - 'NRC_long': Uses the 10 CFR 61.55 limits for long-lived radionuclides
+        - 'NRC_short_A': Uses the 10 CFR 61.55 class A limits for short-lived
+          radionuclides
+        - 'NRC_short_B': Uses the 10 CFR 61.55 class B limits for short-lived
+          radionuclides
+        - 'NRC_short_C': Uses the 10 CFR 61.55 class C limits for short-lived
+          radionuclides
+    metal : bool, optional
+        Whether or not the material is in metal form (only applicable when
+        method is 'NRC')
```

Comment:
```suggestion
        method is 'NRC_*')
```
Technically since `'NRC'` is no longer a method name maybe consider some wildcard indicator or other explanation.